src/reducers/Grid.js

```diff
@@ -20,8 +20,25 @@ const updateBlock = (state, { row, column, machineType }) => ({
   ],
 })
 
+const deletePosition = (gridValues, { row, column }) => {
+  const oldGridValues = gridValues
+
+  oldGridValues[row] = oldGridValues[row].map((value, index) => {
```

Comment:
``` 
oldGridValues[row] = oldGridValues[row].map((value, index) => index === column ? { type: Empty } : value 
```

Reply:
Yeah, I think that's more readable, but there is still another method that looks pretty awful. 
```
const updateBlock = (state, { row, column, machine }) => ({
  ...state,
  gridValues: [
    ...state.gridValues.slice(0, row),
    [
      ...state.gridValues[row].slice(0, column),
      { type: machine.name, data: machine, items: [] },
      ...state.gridValues[row].slice(column + 1),
    ],
    ...state.gridValues.slice(row + 1),
  ],
})
```
I think I'm going to change both of them, Do you agree?

Reply:
fixed on commit https://github.com/PracticaDS/pdes-tp-team_prog/pull/85/commits/7b6e28adfc0566e88cfcbdb12a8bfa74f6031473